homeassistant/components/intellifire/config_flow.py

```diff
@@ -4,60 +4,90 @@
 from typing import Any
 
 from aiohttp import ClientConnectionError
-from intellifire4py import IntellifireAsync
+from intellifire4py import AsyncUDPFireplaceFinder, IntellifireAsync
 import voluptuous as vol
 
 from homeassistant import config_entries
 from homeassistant.const import CONF_HOST
-from homeassistant.core import HomeAssistant
 from homeassistant.data_entry_flow import FlowResult
 
-from .const import DOMAIN
+from .const import DOMAIN, LOGGER
 
 STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
 
 
-async def validate_input(hass: HomeAssistant, host: str) -> str:
+async def validate_host_input(host: str) -> str:
     """Validate the user input allows us to connect.
 
     Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
     """
     api = IntellifireAsync(host)
     await api.poll()
-
+    ret = api.data.serial
+    LOGGER.info("Found a fireplace: %s", ret)
     # Return the serial number which will be used to calculate a unique ID for the device/sensors
-    return api.data.serial
+    return ret
 
 
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for IntelliFire."""
 
-    VERSION = 1
+    VERSION = 2
 
-    async def async_step_user(
-        self, user_input: dict[str, Any] | None = None
-    ) -> FlowResult:
-        """Handle the initial step."""
-        if user_input is None:
-            return self.async_show_form(
-                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
-            )
+    def __init__(self):
+        """Initialize the Config Flow Handler."""
+        self._discovered_host: str = ""
```

Comment:
`_discovered_host` will be used by the future PR for DHCP discovery as well as in this PR via the UDP based discovery.

Reply:
```
homeassistant/components/flux_led/config_flow.py:                return await self.async_step_pick_device()
homeassistant/components/flux_led/config_flow.py:    async def async_step_pick_device(
homeassistant/components/steamist/config_flow.py:    async def async_step_pick_device(
homeassistant/components/steamist/config_flow.py:                return await self.async_step_pick_device()
homeassistant/components/tplink/config_flow.py:                return await self.async_step_pick_device()
homeassistant/components/tplink/config_flow.py:    async def async_step_pick_device(
homeassistant/components/wiz/config_flow.py:    async def async_step_pick_device(
homeassistant/components/wiz/config_flow.py:                return await self.async_step_pick_device()
homeassistant/components/yeelight/config_flow.py:                return await self.async_step_pick_device()
homeassistant/components/yeelight/config_flow.py:    async def async_step_pick_device(self, user_input=None):
```

Here are some examples of how to handle multiple discoveries

Reply:
Thanks... I'll make some changes and push an update.

Reply:
I've got the code written - just waiting on having time to verify the test coverage

Reply:
![image](https://user-images.githubusercontent.com/6491743/155914754-b8cc464f-00ae-47a2-bfc4-5b4fec4656ed.png)